src/user-post/service/post.service.ts

```diff
@@ -0,0 +1,188 @@
+          userIdToken: userId,
+        },
+      })
+      .catch(() => {
+        throw new Error('Something goes wrong');
+      });
+
+    const userPosts: Posts[] = await this.postsRepository.find({
+      where: {
+        userId: getUser.id,
+      },
+      relations: {
+        user: true,
+      },
+      take: 5,
+    });
+    return userPosts;
+  }
+  async getAllPosts(): Promise<Posts[]> {
+    const allPosts: Posts[] = await this.postsRepository.find({
+      relations: {
+        user: true,
+      },
+      take: 5,
+    });
+    return allPosts;
+  }
+  async getMorePosts(length: number): Promise<Posts[]> {
+    const morePosts: Posts[] = await this.postsRepository.find({
+      relations: {
+        user: true,
+      },
+      skip: length,
+      take: 5,
+    });
+    return morePosts;
+  }
+  async getAllPostsLength(): Promise<number> {
+    const allPosts: Posts[] = await this.postsRepository.find();
+    return allPosts.length;
+  }
+  async getAllUserPostsLength(userId: string) {
+    const getUser: User = await this.userRepository.findOneBy({
+      userIdToken: userId,
+    });
+    const allPosts: Posts[] = await this.postsRepository.find({
+      where: {
+        userId: getUser.id,
+      },
+    });
+    return allPosts.length;
+  }
+  async getMoreUserPosts(length: number, uId: string): Promise<Posts[]> {
+    console.log(length);
+    try {
+      const currentUser: User = await this.userRepository.findOne({
+        where: {
+          userIdToken: uId,
+        },
+      });
+
+      const moreUserPosts: Posts[] = await this.postsRepository.find({
+        where: {
+          userId: currentUser.id,
+        },
+        relations: {
+          user: true,
+        },
+        skip: length,
+        take: 5,
+      });
+      return moreUserPosts;
+    } catch (e) {
+      throw new Error(e.message);
+    }
+  }
+
+  async editPost(
+    editFormData: EditPostDto,
+    photo: Express.Multer.File,
+    postId: number,
+    uId: string,
+  ): Promise<[Posts[], string]> {
+    try {
+      const getUserID: User = await this.userRepository.findOneBy({
+        userIdToken: uId,
+      });
+      const getCurrentPost: Posts = await this.postsRepository.findOneBy({
+        id: postId,
+        userId: getUserID.id,
+      });
+      const oldPostPhoto: string = getCurrentPost.photo;
+      getCurrentPost.title = editFormData.title;
+      getCurrentPost.description = editFormData.description;
+      if (editFormData.delete === 'true') {
+        getCurrentPost.photo = '';
+      }
+      if (photo) {
+        getCurrentPost.photo = await this.firebaseService.uploadFile(photo);
+      }
+      await this.postsRepository.save(getCurrentPost);
+      const updated: Posts[] = await this.postsRepository.find({
+        where: {
+          userId: getUserID.id,
+        },
+        relations: { user: true },
+        order: {
+          updatedAt: 'DESC',
+        },
+      });
+      return [updated, oldPostPhoto];
```

Comment:
Hmmm, why?

Reply:
That part has changed too. In the old version of my code, photo deletion from Firebase Storage was handled on the frontend, now, I've figured out how to do it on the backend, eliminating the need to send the old photo rl.